src/plugins/synced-lyrics/providers/YTMusic.ts

```diff
@@ -0,0 +1,172 @@
+import { LyricProvider, LyricResult, SearchSongInfo } from '../types';
+
+const headers = {
+  Accept: 'application/json',
+  'Content-Type': 'application/json',
+};
+
+const client = {
```

Comment:
use the youtubei.js instead of this

Reply:
I did spend a considerable amount of time trying to do just that.

But it doesn't allow me to modify the client information.

Reply:
- If `youtubei.js` is impossible to use, you must have a way to use the user's YouTube session (using cookies).
  - In countries like South Korea, YTM features (including the `/next` endpoint) such as lyrics, song information, song playback, and search are unavailable without subscribing to YouTube Premium.
    - If the account does not have YouTube Premium, or if the session is not signed in, the response will always be empty

Reply:
I will give it a try

Reply:
I used UrbanVPN to connect to a Seoul server, and visited yt music from incognito and stuff like search and plain lyrics worked.
Does youtube not block those endpoints for VPNs?

Reply:
Sadly I don't have an immediate solution, but I do have an idea in mind to resolve this issue forever:tm:

I am thinking of making a backend that fetches lyrics on demand and then caches them, to act like a proxy but just for lyrics.

Reply:
> I used UrbanVPN to connect to a Seoul server, and visited yt music from incognito and stuff like search and plain lyrics worked. Does youtube not block those endpoints for VPNs?

That IP band may have been recognized as a country other than South Korea.

Reply:
I will implement a cookie parser for youtubei when I have time.

Reply:
The issue is not the cookie, it's having an authenticated session with the iOS client

Unless I've missed smth and you can reuse the web cookie for the iOS client.